### PR TITLE
Move InserterButton from components package to block-editor package

### PR DESCRIPTION
**Pull request by derekblank**
## What?
This PR moves the mobile `InserterButton` from the components package to the block-editor package.

## Why?
There is an ongoing effort to remove block-editor package dependencies in native files from the components package in order to eliminate cyclic dependencies:

* https://github.com/WordPress/gutenberg/issues/52692
* https://github.com/WordPress/gutenberg/issues/56072
* https://github.com/WordPress/gutenberg/issues/27751

## How?
Moves code and code references to `InserterButton` from the components package to the block-editor package.

## Testing Instructions
1. Create a post and use the Block Inserter button (bottom left corner of screen)
2. Button should function as normal compared to production

### CODE REVIEW

File: packages/block-editor/src/components/block-variation-picker/index.native.js
Comment (fluiddot) [inline packages/block-editor/src/components/block-variation-picker/index.native.js]: Since this component is in the same package, we could import it using a relative path.

File: packages/block-editor/src/components/inserter-button/sparkles.js
Comment (fluiddot) [inline packages/block-editor/src/components/inserter-button/sparkles.js:4]: It's true that these components are exported in the `components` package ([reference](https://github.com/WordPress/gutenberg/blob/c86154f838cef387785a948e30a5a7c7cbbae47f/packages/components/src/index.native.js#L1-L10)) but they are originally defined in `primitives` package. Do you foresee any benefit if we import from `components` package instead of `primitives`?

File: packages/block-editor/src/components/block-types-list/index.native.js
Comment (fluiddot) [inline packages/block-editor/src/components/block-types-list/index.native.js]: Since this component is in the same package, we could import it using a relative path.

File: packages/components/src/index.native.js
Comment (fluiddot) [inline packages/components/src/index.native.js:113]: I saw that the `InserterButton` component is used in the `block-experiments` repository ([reference](https://github.com/Automattic/block-experiments/blob/184b3dd7f1de5a405cff86c7cf90232c5bea4847/blocks/layout-grid/src/grid/variation-control/index.native.js#L18)). We should update it as a follow-up on this PR.

**Review by fluiddot — APPROVED** [review]
LGTM 🎊 !

I added some minor comments that shouldn't block the merge of the PR. Let me know if you'd like to address them, otherwise feel free to merge the PR as is.

File: packages/block-editor/src/components/inserter-button/sparkles.js
Comment (derekblank) [reply packages/block-editor/src/components/inserter-button/sparkles.js:4]: Importing these components from `@wordpress/primitives` throws a linter error in this case. It appears that in most packages, these components are imported from `@wordpress/primitives` ([#](https://github.com/search?q=repo%3AWordPress%2Fgutenberg+import+%7B+SVG%2C+Path+%7D+from+%27%40wordpress%2Fprimitives%27%3B&type=code)), but in the `block-editor` and `block-library` packages, they are imported from `@wordpress/components` ([#](https://github.com/search?q=repo%3AWordPress%2Fgutenberg+import+%7B+SVG%2C+Path+%7D+from+%27%40wordpress%2Fcomponents%27%3B&type=code)). I was somewhat curious if there was a particular reason for this, but didn't note one. This change does match the import convention of other `block-editor` components, however.

File: packages/components/src/index.native.js
Comment (derekblank) [reply packages/components/src/index.native.js:113]: Good catch. PR opened here: https://github.com/Automattic/block-experiments/pull/338

**Comment by derekblank** [comment]
> LGTM 🎊 !
> 
> I added some minor comments that shouldn't block the merge of the PR. Let me know if you'd like to address them, otherwise feel free to merge the PR as is.

@fluiddot Thanks for the review - I updated the relative import paths and linked the [block-experiments PR](https://github.com/Automattic/block-experiments/pull/338). I also opened a [companion Gutenberg Mobile PR](https://github.com/wordpress-mobile/gutenberg-mobile/pull/6414), but I'm not sure it really needs one (or at least, I don't think it needs an update to the RELEASE-NOTES.txt in Gutenberg Mobile). WDYT?

File: packages/block-editor/src/components/inserter-button/sparkles.js
Comment (fluiddot) [reply packages/block-editor/src/components/inserter-button/sparkles.js:4]: Oh, I didn't know that importing directly from `@wordpress/primitives` in the `block-editor` package produced a linter error. OK, let's stick with this change then, thanks for elaborating 🙇 !

**Comment by fluiddot** [comment]
> I also opened a [companion Gutenberg Mobile PR](https://github.com/wordpress-mobile/gutenberg-mobile/pull/6414), but I'm not sure it really needs one (or at least, I don't think it needs an update to the RELEASE-NOTES.txt in Gutenberg Mobile). WDYT?

@derekblank The only reason for creating the companion PR is due to the update in the `block-experiments` repository (https://github.com/Automattic/block-experiments/pull/338). Once it's merged we'd need to update the reference in Gutenberg Mobile, so we can use the companion PR to align the changes. WDYT?

**Comment by derekblank** [comment]
@fluiddot I agree. I'll update the block-experiments ref in the Gutenberg Mobile PR once the block-experiments PR is approved/merged. 👍

**Comment by fluiddot** [comment]
> @fluiddot I agree. I'll update the block-experiments ref in the Gutenberg Mobile PR once the block-experiments PR is approved/merged. 👍

@derekblank  I've approved all the PRs, feel free to start updating the references and merge them. Thanks 🙇 !

**Pull request merged**
